versioned_hdf5/wrappers.py

```diff
@@ -531,8 +520,32 @@ def _astype_impl(self, dtype: np.dtype, writeable: bool) -> MutableArrayProtocol
         """Return self._buffer as a new dtype. Return a view if possible."""
 
 
-# Note: BufferMixin methods override those from Dataset
-class InMemoryDataset(BufferMixin, Dataset):
+class CompressionMixin:
+    name: str
+    parent: InMemoryGroup
+
+    @property
+    def compression(self):
+        _, basename = posixpath.split(self.name)
+        return self.parent.compression[basename]
```

Comment:
```suggestion
        return self.parent.compression[posixpath.basename(self.name)]
```

Any reason not to use posixpath.basename?